### ISSUE TITLE
Add Support for Zip packages

### ISSUE THREAD

**Issue by kayone** [appget/appget#4]
Right now appget is limited to supporting applications with installers. however many applications don't have installers and are shipped as xcopyable compressed files.

Arguably a zip package installer would provide more value for users since using non-installable apps as user is more annoying,

Steps:

- downloading the zip package
- extracting and placing the application somewhere to be run
- if the application contains more than one .exe figuring out which one needs to be run
- every time the app needs to be run the user has to manually navigate to the folder or
- manually create a shortcut/pin for the app
- if the app needs to run on startup then things get even more complicated

AppGet could easily solve pretty much all those issues,

- downloading and extracting the app to a set target is trivial
- if the application contains more than one .exe file it could be specified in the manifest
- we could create a shim to a folder that has been added to the `PATH`, similar to how node installs global packages, that way the user could run the app from any folder from any shell
- Once we know the main executable of a package getting it pinned to taskbar/start/desktop/startup is trivial

Prototype template:

```
id: vlc
name: VLC media player
version: 3.0.3
home: https://videolan.org/
license: GNU Lesser General Public License v2.1+
installMethod: NSIS
installers:
- location: https://get.videolan.org/vlc/3.0.3/win32/vlc-3.0.3-win32.zip
- location: https://get.videolan.org/vlc/3.0.3/win64/vlc-3.0.3-win64.zip
  architecture: x64
```

#### Notes:
- install method could be omitted since .zip extension makes it implicit.
- Could main executable be implied since it exactly matches the package name? or maybe we should be explicit?

**Comment by eberkund** [comment]
I wouldn't mind working on this, do you want some help?

**Comment by kayone** [comment]
Services are something that I've thought about a lot. and it's definitely something that I would love to include.

I would love some help, I think the first thing we need to figure out is how would the manifest look like.

**Comment by eberkund** [comment]
Well, the prototype you posted looks like a good start. I think services, shortcuts are a bit more involved so maybe that could be done after implementing CLI tools first.

I think for adding to the $PATH we could either append the entire installation folder or add a key to the manifest to specify which executables need to be added to the $PATH.

**Comment by eberkund** [comment]
Yeah that makes sense, however in some cases I think it makes sense to add to PATH. For example, if you install NPM or Composer or Golang then those are package managers themselves which have their own bin directories which need to be in the PATH. The Windows installers for these applications add to PATH right now.

**Comment by liamdawson** [comment]
On the `PATH` discussion, I've had my `PATH` variable become too long in Windows, to the point I couldn't edit it without extra workarounds. Maybe support having **one** directory where path-suitable executables can go, like Homebrew does? (`/usr/local/bin`)
